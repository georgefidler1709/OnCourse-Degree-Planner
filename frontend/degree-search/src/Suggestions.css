/* Style the degree suggestion units */
.suggestion {
  border: 1px solid #575756;
  border-radius: 10px;
  background-color: #c2eced;
  padding: 10px;
  align-self: center;
}

.suggestion-title {
  pointer-events: none;
}

.suggestion-desc {
  pointer-events: none;
}