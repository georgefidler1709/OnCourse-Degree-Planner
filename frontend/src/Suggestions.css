/* Style the degree suggestion units */
.suggestion {
  border: 1px solid #575756;
  border-radius: 10px;
  background-color: #c2eced;
  padding: 10px;
  align-self: center;
}

.suggestion-code {
  pointer-events: none;
}

.suggestion-name {
  pointer-events: none;
}