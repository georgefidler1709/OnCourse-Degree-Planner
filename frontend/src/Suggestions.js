import React from "react";
import Button from "react-bootstrap/Button";

function Suggestions(props) {
  const options = props.degrees.map((r,i) => (
    <div key={i}>
      <Button
        variant="light"
        className="suggestion"
        id={r.id}
        value={r.name}
        onClick={event => console.log(event.target.value)}
      >
        <h1 className="suggestion-title">{r.code}</h1>
        <p className="suggestion-desc">{r.name}</p>
      </Button>
      <br />
    </div>
  ));

  return <div>{options}</div>;
}

export default Suggestions;
