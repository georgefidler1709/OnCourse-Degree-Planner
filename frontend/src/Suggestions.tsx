import React, {MouseEvent} from "react";
import Button from "react-bootstrap/Button";
import SuggestionInfoHover from "./SuggestionInfoHover"

import {Position} from './Types'
import {SimpleDegrees} from './Api'

const year : string = "2020"
const handbook : string = `https://www.handbook.unsw.edu.au/undergraduate/specialisations/${year}/`
const placement : Position = "right"

function Suggestions(props: {degrees: SimpleDegrees}) {
  const options = props.degrees.map((r,i) => (
    <SuggestionInfoHover
      content={
        <a href={handbook + r.id.toString()}>More Info</a>
      }
      placement={placement}
      delay={200}
      key={r.id}
    >
      <Button
        variant="light"
        className="suggestion"
        id={r.id.toString()}
        value={r.name}
        onClick={(event: MouseEvent<HTMLButtonElement>) => console.log(event.currentTarget.value)}
      >
        <h1 className="suggestion-code">{r.id}</h1>
        <p className="suggestion-name">{r.name}</p>
      </Button>
    </SuggestionInfoHover>
  ));

  return <div>{options}</div>;
}

export default Suggestions;
