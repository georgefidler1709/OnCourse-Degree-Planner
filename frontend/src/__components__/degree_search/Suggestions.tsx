import React, {MouseEvent} from "react";
import Button from "react-bootstrap/Button";
import SuggestionInfoHover from "./SuggestionInfoHover"
import { useHistory } from "react-router-dom";
import {Position} from '../../Types'
import {SimpleDegrees} from '../../Api'


function Suggestions(props: {degrees: SimpleDegrees}) {

  const year : string = "2020"
  const handbook : string = `https://www.handbook.unsw.edu.au/undergraduate/programs/${year}/`
  const placement : Position = "right"
  let history = useHistory();

  function handleClick(event: MouseEvent<HTMLButtonElement>) {
    history.push("/" + event.currentTarget.id.toString())
    console.log(event.currentTarget);
    console.log(event.currentTarget.value);
    console.log(event.currentTarget.id.toString());
  }
  
  const options = props.degrees.map((r,i) => (
    <SuggestionInfoHover
      content={
        <a href={handbook + r.id}>More Info</a>
      }
      placement={placement}
      delay={200}
      key={r.id}
    >
      <Button
        variant="light"
        className="suggestion"
        id={r.id}
        value={r.name}
        onClick={handleClick}
      >
        <h1 className="suggestion-code">{r.id}</h1>
        <p className="suggestion-name">{r.name}</p>
      </Button>
    </SuggestionInfoHover>
    
  ));

  return <div className="suggestion-container">{options}</div>;
}

export default Suggestions;
