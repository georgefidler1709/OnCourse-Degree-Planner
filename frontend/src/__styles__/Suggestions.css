/* Style the degree suggestion units */
.suggestion {
  border: 1px solid #575756;
  border-radius: 10px;
  background-color: #c2eced;
  padding: 10px;
  margin-bottom: 8px;
  width: 220px;
  
}

.suggestion-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.suggestion-code {
  pointer-events: none;
}

.suggestion-name {
  pointer-events: none;
}

.suggestion-info {
  transition: all 2s ease-in;
}

