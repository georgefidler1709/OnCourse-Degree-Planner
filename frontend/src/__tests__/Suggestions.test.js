import React from 'react'
import { shallow, mount } from 'enzyme';
import Suggestions from '../__components__/degree_search/Suggestions';

console.warn = jest.fn();

const mockDegrees = [
	{ degree: {
	  "id": "3778", 
	  "name": "Computer Science"
	} }, 
	{ degree: {
	  "id": "7001", 
	  "name": "Science"
	} }, 
  ]

jest.mock("react-router-dom", () => ({
	useHistory: jest.fn(() => {
		return {
			push: jest.fn()
		}
	})
  }))

describe('Rendering a degree suggestion', () => {
    it('renders correctly', () => {
	  const wrapper = shallow(<Suggestions degrees={mockDegrees} year={2020} />);
	  expect(wrapper).toMatchSnapshot();
	});

	it('puts the code of provided degrees as titles', () => {
		const wrapper = shallow(<Suggestions degrees={mockDegrees} year={2020}/>);
		
		expect(wrapper.html().includes(">3778</h1>")).toBeTruthy()
	})

	it('renders degree suggestions in the order they are provided', () => {
		const wrapper = shallow(<Suggestions degrees={mockDegrees} year={2020}/>);
		const code1 = wrapper.html().indexOf(">3778 COMPA1</h1>")
		const code2 = wrapper.html().indexOf(">3707 SENGAH</h1>")
		expect(wrapper.html()).toBe('f');
	})	
});
