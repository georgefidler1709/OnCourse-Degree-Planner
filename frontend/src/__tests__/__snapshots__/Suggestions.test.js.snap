// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering a degree suggestion renders correctly 1`] = `
<styled.div>
  <SuggestionInfoHover
    content={
      <a
        href="https://www.handbook.unsw.edu.au/undergraduate/programs/2020/3778"
        rel="noopener noreferrer"
        target="_blank"
      >
        More Info
      </a>
    }
    delay={100}
    infoSize={
      Object {
        "fontSize": "25px",
      }
    }
    key="3778"
    placement="right"
  >
    <Styled(Button)
      id="3778"
      onClick={[Function]}
      value="3778"
      variant="light"
    >
      <styled.h1>
        3778
      </styled.h1>
      <styled.p />
    </Styled(Button)>
  </SuggestionInfoHover>
  <SuggestionInfoHover
    content={
      <a
        href="https://www.handbook.unsw.edu.au/undergraduate/programs/2020/7001"
        rel="noopener noreferrer"
        target="_blank"
      >
        More Info
      </a>
    }
    delay={100}
    infoSize={
      Object {
        "fontSize": "25px",
      }
    }
    key="7001"
    placement="right"
  >
    <Styled(Button)
      id="7001"
      onClick={[Function]}
      value="7001"
      variant="light"
    >
      <styled.h1>
        7001
      </styled.h1>
      <styled.p />
    </Styled(Button)>
  </SuggestionInfoHover>
</styled.div>
`;
